syntax = "proto3";

package omnipricex.auth;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

// Authentication Service Definition
service AuthService {
    rpc RegisterUser(RegisterUserRequest) returns (UserResponse);
    rpc LoginUser(LoginUserRequest) returns (LoginUserResponse);
    rpc VerifyToken(VerifyTokenRequest) returns (VerifyTokenResponse);
    rpc RefreshToken(RefreshTokenRequest) returns (LoginUserResponse);
    rpc LogoutUser(LogoutUserRequest) returns (google.protobuf.Empty);
    rpc GetUserProfile(GetUserProfileRequest) returns (UserResponse);
    rpc UpdateUserProfile(UpdateUserProfileRequest) returns (UserResponse);
}

// Messages
message User {
    string id = 1;
    string username = 2;
    string email = 3;
    string full_name = 4;
    bool is_active = 5;
    bool is_staff = 6;
    google.protobuf.Timestamp created_at = 7;
    google.protobuf.Timestamp updated_at = 8;
}

// Request/Response messages
message RegisterUserRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    string full_name = 4;
}

message LoginUserRequest {
    string email = 1;
    string password = 2;
}

message LoginUserResponse {
    string access_token = 1;
    string refresh_token = 2;
    User user = 3;
}

message VerifyTokenRequest {
    string token = 1;
}

message VerifyTokenResponse {
    bool is_valid = 1;
    string user_id = 2;
    string email = 3;
}

message RefreshTokenRequest {
    string refresh_token = 1;
}

message LogoutUserRequest {
    string user_id = 1;
}

message GetUserProfileRequest {
    string user_id = 1;
}

message UpdateUserProfileRequest {
    string user_id = 1;
    optional string username = 2;
    optional string email = 3;
    optional string full_name = 4;
}

message UserResponse {
    bool success = 1;
    string message = 2;
    User user = 3;
}
